playground/CosmosEndToEnd/CosmosEndToEnd.ApiService/Program.cs

```diff
@@ -8,8 +8,9 @@
 var builder = WebApplication.CreateBuilder(args);
 
 builder.AddServiceDefaults();
-builder.AddAzureCosmosDatabase("db");
-builder.AddAzureCosmosContainer("entries");
+builder.WithAzureCosmosDatabase("db")
+    .AddContainer("entries")
```

Comment:
I wonder if this should be called `WithKeyedContainer`.

1. We use "With" when you return the same builder, "Add" when you return the new thing being added.
2. Adding "Keyed" makes it obvious that the service will be keyed.

Reply:
And the root method should be `AddAzureCosmosDatabase`.

Reply:
> And the root method should be AddAzureCosmosDatabase.

I like the prefix `With` for this API because it gives a better clue that you're registering containers within the context of a specific database. 

Also, with:

> "Add" when you return the new thing being added.

The `WithAzureCosmosDatabase` doesn't register any services currently so using the `Add` prefix would violate this convention.

Reply:
My thinking was that we'd have:

```C#
builder.AddAzureCosmosDatabase("db")
    .WithKeyedContainer("entries")
    .WithKeyedContainer("users");
```

And it would register 3 services:

1. The unkeyed Cosmos `Database`.
2. Two keyed Cosmos `Container` services.

And if I want to register keyed Databases, I would use:

```C#
builder.AddKeyedAzureCosmosDatabase("db")
    .WithKeyedContainer("entries")
    .WithKeyedContainer("users");
```

Reply:
> And it would register 3 services:

> The unkeyed Cosmos Database.
> Two keyed Cosmos Container services.

I find this a little confusing given that our convention in the other types has been that they only register the service that they are associated with. For example, currently when you call `AddKeyedAzureCosmosDatabase`, you don't also get a `CosmosClient` registered as well.